### PR TITLE
Addition of more comments to MyEgress and MyVerifyChecksum in basic_tunnel.p4

### PR DESCRIPTION
**Pull request by Abhinavcode13**
- Added detailed comments for MyEgress and MyVerifyChecksum blocks.

### CODE REVIEW

**Comment by Abhinavcode13** [comment]
Any suggestions or feedback is appreciated for this. @jafingerhut

File: exercises/basic_tunnel/basic_tunnel.p4 [hunk 1]
```diff
@@ -87,7 +87,24 @@ parser MyParser(packet_in packet,
...
 
 control MyVerifyChecksum(inout headers hdr, inout metadata meta) {
-    apply {  }
+    apply { 
+        /*
+        Control function for checksum verification.
+       
+        This function is responsible for the MyVerifyChecksum control block in P4, 
+        which checks and drops IPv4 packets with incorrect checksums. It uses inout
+        headers hdr to manipulate packet headers and inout metadata meta to store
```
Comment (jafingerhut) [inline exercises/basic_tunnel/basic_tunnel.p4:96]: This block is responsible for checking checksums in input packet headers, but it is _not_ responsible for dropping packets with incorrect checksums.  See https://github.com/p4lang/behavioral-model/blob/main/docs/simple_switch.md#restrictions-on-code-in-the-verifychecksum-control for details.

File: exercises/basic_tunnel/basic_tunnel.p4 [hunk 2]
```diff
@@ -144,7 +161,26 @@ control MyIngress(inout headers hdr,
...
                  inout metadata meta,
                  inout standard_metadata_t standard_metadata) {
-    apply {  }
+    apply { 
+        /*
+        Control function for egress processing.
+       
+        This function is responsible for handling outbound packet operations 
+        such as modifying headers, managing metadata, and making forwarding 
+        choices.
```
Comment (jafingerhut) [inline exercises/basic_tunnel/basic_tunnel.p4:170]: In the V1model architecture, all choices of the packet's output port must be made in the ingress control.  The egress control cannot change the packet's output port.  See https://github.com/p4lang/behavioral-model/blob/main/docs/simple_switch.md for details.

File: exercises/basic_tunnel/basic_tunnel.p4 [hunk 3]
```diff
@@ -144,7 +161,26 @@ control MyIngress(inout headers hdr,
...
+        Parameters: 
+        - inout headers hdr: "hdr" is used to manipulate packet headers. 
+        - inout metadata meta: "meta" stores metadata during packet processing.
+        - inout standard_metadata_t standard_metadata: Mutable standard packet 
+          metadata accessible for modification in packet processing stages.
+        Note : inout indicates that the parameter is an input and an ouput.
+
+        TODO: Implement the logic to modify packet headers based on policy rules 
+        and determine the egress port for forwarding based on the destination IP 
+        addr.
```
Comment (jafingerhut) [inline exercises/basic_tunnel/basic_tunnel.p4:181]: Please remove the "and determine eht egress port ..." part of this sentence.  If the ingress control is missing code to do this, it is appropriate to have a comment to add such code to the ingress control, but as mentioned in another comment, the egress control cannot change the packet's output port.

File: exercises/basic_tunnel/basic_tunnel.p4 (hunk 1 above)
Comment (Abhinavcode13) [reply exercises/basic_tunnel/basic_tunnel.p4]: Okay, I fixed it. By the way, then which block takes care of dropping packets with incorrect checksums?

File: exercises/basic_tunnel/basic_tunnel.p4 (hunk 3 above)
Comment (Abhinavcode13) [reply exercises/basic_tunnel/basic_tunnel.p4:182]: So what we can add here in todo for this block?

Comment (jafingerhut) [reply exercises/basic_tunnel/basic_tunnel.p4:182]: I guess more importantly is: do we need these changes at all?  Is there something in the exercise that recommends changing these parts of the basic_tunnel.p4 program at all?

Comment (Abhinavcode13) [reply exercises/basic_tunnel/basic_tunnel.p4:182]: I believe that as well, but I felt that I should complete it for basictunnel.p4 in considering PR #572 

Nevermind , We can close this PR as of now.

**Comment by jafingerhut** [comment]
Sure, the difference is: in that other P4 program, those comments were pointing out places in the code that needed to be changed in order to complete the exercise.

**Comment by Abhinavcode13** [comment]
Not necessary as of now, I'll close this one.

**Pull request closed without merging**